src/blob/core.rs

```diff
@@ -183,31 +183,54 @@ impl Blob {
         debug!("blob read any");
         let entry = self
             .get_any_entry(key, meta)
-            .await?
+            .await
+            .unwrap()
             .ok_or_else(|| Error::from(ErrorKind::RecordNotFound))?;
         debug!("blob read any entry found");
         let buf = entry
             .load()
             .await
-            .with_context(|| format!("failed to read key {:?} with meta {:?}", key, meta))?
+            .with_context(|| format!("failed to read key {:?} with meta {:?}", key, meta))
+            .unwrap()
             .into_data();
         debug!("blob read any entry loaded bytes: {}", buf.len());
         Ok(buf)
     }
 
     #[inline]
-    pub(crate) fn read_all<'a>(&'a self, key: &'a [u8]) -> Entries<'a> {
-        self.index.get_entries(key, self.file.clone())
+    pub(crate) async fn read_all(&self, key: &[u8]) -> Result<Option<Vec<Entry>>> {
+        let headers = self.index.get_all(key).await?;
+        Ok(headers.map(|h| {
+            debug!("blob core read all {} headers", h.len());
+            Self::headers_to_entries_with_meta(h, self.file.clone())
```

Comment:
I think using method here is redundant

Reply:
I use it in other place [here](https://github.com/qoollo/pearl/pull/56/files#diff-f2bad8d74f50db884e82f478e0f8ceebR242)
Nested closures are hard to read (: